### PR TITLE
Add usage example for MaxPool2D

### PR DESCRIPTION
**Pull request by Joseph-Rance**
Add usage example for MaxPool2D

### CODE REVIEW

**Comment by Joseph-Rance** [comment]
Hi, @rthadur, It has been a couple of days since this has been last looked at. Is there anything I need to do and is there any way I can expedite the review of this PR as I am doing it as part of Google CodeIn 2019.

**Comment by rthadur** [comment]
@dynamicwebpaige gentle ping for review.
cc @mihaimaruseac

File: tensorflow/python/keras/layers/pooling.py [hunk 1]
```diff
@@ -370,6 +370,13 @@ class MaxPooling2D(Pooling2D):
             [[10.],
              [11.],
              [12.]]]], dtype=float32)>
+             
+  Usage Example:
+  
+  >>> model = tf.keras.models.Sequential()
+  >>> model.add(tf.keras.layers.Conv2D(32, kernel_size=(3, 3), 
+  ...    input_shape=(28,28,1)))
+  >>> model.add(tf.keras.layers.MaxPooling2D(pool_size=(2, 2)))
```
Comment (mihaimaruseac) [inline tensorflow/python/keras/layers/pooling.py]: Shouldn't this print something?

Usually for a doctest you want to have some code and some output so when the test runs it checks that the output during the test is the same as the one that is on the file.

Probably create a toy input before creating the model and then run `model.fit` on the toy input and print the output tensor

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/python/keras/layers/pooling.py (hunk 1 above)
Comment (Joseph-Rance) [reply tensorflow/python/keras/layers/pooling.py]: HI @mihaimaruseac, thanks for the really quick reply. I have made some changes so that an output is given but I'm not sure that I have done the correct thing here. Should I be making a prediction from the trained model after the code I added?

Comment (mihaimaruseac) [reply tensorflow/python/keras/layers/pooling.py]: Yes, please.

Also, instead of having the image created from random, please make it a small constant using `tf.constant`. This way, when the random test seed changes the test can still pass.

Comment (Joseph-Rance) [reply tensorflow/python/keras/layers/pooling.py]: Hi, thanks for the fast reply. How should I be making predictions as model.predict returns a numpy array instead of a tensor?

Comment (mihaimaruseac) [reply tensorflow/python/keras/layers/pooling.py]: You can print the numpy array too.

Comment (Joseph-Rance) [reply tensorflow/python/keras/layers/pooling.py]: I used tf.constant for the input image, but the code still outputs different array each time I run it. I have added seeds and turned of shuffle and it still outputs different each time. Do you know what I need to do to fix this? This is what I have at the moment:

>>> input_image = tf.constant([[[[1.], [1.], [2.], [4.], [2.], [4.], [2.]],
...                            [[2.], [2.], [3.], [2.], [2.], [1.], [2.]],
...                            [[4.], [1.], [1.], [1.], [1.], [2.], [2.]],
...                            [[2.], [2.], [1.], [4.], [2.], [3.], [4.]],
...                            [[1.], [4.], [1.], [1.], [2.], [3.], [2.]],
...                            [[1.], [4.], [2.], [3.], [1.], [2.], [3.]],
...                            [[3.], [4.], [1.], [2.], [3.], [1.], [4.]]]]) 
>>> output = tf.constant([[[[1], [0]],
...                       [[0], [1]]]]) 
>>> model = tf.keras.models.Sequential()
>>> model.add(tf.keras.layers.Conv2D(1, kernel_size=(3, 3), 
...    input_shape=(7,7,1)))
>>> model.add(tf.keras.layers.MaxPooling2D(pool_size=(2, 2)))
>>> model.compile('adam', 'mean_squared_error')
>>> model.fit(input_image, output, steps_per_epoch=1, 
...    shuffle=False)
>>> model.predict(input_image, steps=1)
array([[[[1.7140688 ], [1.3748764 ]],
        [[0.08504223], [0.8180907 ]]]], dtype=float32)

Thanks for the help

Comment (Joseph-Rance) [reply tensorflow/python/keras/layers/pooling.py]: @mihaimaruseac I also have another PR open at the moment (#35715 ) in which some of the changes here may apply. Could you advise whether they do or of it is different for that one?

**Comment by Joseph-Rance** [comment]
@mihaimaruseac I've updated it so that it prints the shape of the output instead of the actual output. Is this ok? If not, how should I be ensuring that the output is always the same?

**Comment by Joseph-Rance** [comment]
@rthadur, Mihai Maruseac mentioned that I would need to print an output the doctest. Do you know whether the current commit has an acceptable output for the doctest?

**Comment by Joseph-Rance** [comment]
@rthadur as mentioned, I have until the 23rd to finish this pr so if it is possible could you give a little guidance on what needs to be done here. Thanks.

**Comment by mihaimaruseac** [comment]
@Joseph-Rance sorry for the delay, we had a 3-days long weekend over here so I minimized amount of time spent with work related things.

You should try to ensure that the output always stays the same. See guidelines at https://www.tensorflow.org/community/contribute/docs_ref

**Comment by Joseph-Rance** [comment]
Hi, thanks for the reply. I'll try to get the output correct based on that documentation. Do you know how I can remove the random element of the .fit() function, though?

**Review by mihaimaruseac — APPROVED** [review]

**Comment by Joseph-Rance** [comment]
I couldn't find a way to get the same output every time with the .fit() function, so I removed the conv2d() layer from the model (as they are not necessary for the usage example) to remove the need for .fit(). Now the code should make the same prediction each time.

**Comment by Joseph-Rance** [comment]
Hi, I just wanted to say thank you for all your help (and putting up with my @ing over the weekend). Joseph
